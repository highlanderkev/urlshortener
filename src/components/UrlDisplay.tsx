import React from "react";
import { URLEntry } from "../urlEntry.model";

type UrlDisplayProps = {
  entry: URLEntry;
};

export default class UrlDisplay extends React.Component<UrlDisplayProps> {
  render() {
    return (
      <div>
        <h3>Original URL: {this.props.entry.url}</h3>
        <h4>Shortened URL: {this.props.entry.short_url}</h4>
      </div>
    );
  }
}
