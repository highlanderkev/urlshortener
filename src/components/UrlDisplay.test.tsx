import React from 'react';
import UrlDisplay from "./UrlDisplay";
import renderer from "react-test-renderer";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()})
describe('Url Display Component', () => {
  it("matches snapshot", () => {
    const urlDisplay = renderer.create(<UrlDisplay entry={{ url: 'Test', short_url: 'Test', slug: '' }} />);
    let tree = urlDisplay.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("displays url entry", () => {
    const display = shallow(<UrlDisplay entry={{ url: 'Test', short_url: 'Test', slug: '' }} />)
    expect(display.text()).toEqual("Original URL: TestShortened URL: Test");
  });
})
