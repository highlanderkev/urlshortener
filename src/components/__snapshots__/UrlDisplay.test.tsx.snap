// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Url Display Component matches snapshot 1`] = `
<div>
  <h3>
    Original URL: 
    Test
  </h3>
  <h4>
    Shortened URL: 
    Test
  </h4>
</div>
`;
